src/processor/FHIRProcessor.ts

```diff
@@ -7,52 +7,72 @@ import {
   ValueSetProcessor,
   ConfigurationProcessor
 } from '.';
-import {
-  ExportableProfile,
-  ExportableExtension,
-  ExportableCodeSystem,
-  ExportableValueSet,
-  ExportableConfiguration,
-  ExportableInvariant
-} from '../exportable';
+import { ExportableConfiguration } from '../exportable';
+import { ConfigurationExtractor } from '../extractor';
+import { Package } from './Package';
 
 export class FHIRProcessor {
   public readonly structureDefinitions: any[] = [];
+  public readonly codeSystems: any[] = [];
+  public readonly valueSets: any[] = [];
+  public readonly implementationGuides: any[] = [];
   public readonly fhir: fhirdefs.FHIRDefinitions;
 
   constructor(fhir: fhirdefs.FHIRDefinitions) {
     this.fhir = fhir;
   }
 
-  process(
-    inputPath: string,
-    existingInvariants: ExportableInvariant[] = []
-  ): (
-    | ExportableProfile
-    | ExportableExtension
-    | ExportableCodeSystem
-    | ExportableValueSet
-    | ExportableConfiguration
-    | ExportableInvariant
-  )[] {
+  register(inputPath: string): void {
     const rawContent = JSON.parse(fs.readFileSync(inputPath, 'utf-8'));
 
     if (rawContent['resourceType'] === 'StructureDefinition') {
       // Profiles and Extensions are both made from StructureDefinitions
       // Invariants may be contained within StructureDefinitions
       this.structureDefinitions.push(rawContent);
-      logger.debug(`Processing contents of ${inputPath} as StructureDefinition.`);
-      return StructureDefinitionProcessor.process(rawContent, this.fhir, existingInvariants);
+      logger.debug(`Registered contents of ${inputPath} as StructureDefinition.`);
     } else if (rawContent['resourceType'] === 'CodeSystem') {
-      logger.debug(`Processing contents of ${inputPath} as CodeSystem.`);
-      return [CodeSystemProcessor.process(rawContent)];
+      this.codeSystems.push(rawContent);
+      logger.debug(`Registered contents of ${inputPath} as CodeSystem.`);
     } else if (rawContent['resourceType'] === 'ValueSet') {
-      logger.debug(`Processing contents of ${inputPath} as ValueSet.`);
-      return [ValueSetProcessor.process(rawContent)];
+      this.valueSets.push(rawContent);
+      logger.debug(`Registered contents of ${inputPath} as ValueSet.`);
     } else if (rawContent['resourceType'] === 'ImplementationGuide') {
-      return [ConfigurationProcessor.process(rawContent)];
+      this.implementationGuides.push(rawContent);
+    }
```

Comment:
Should we have an `else` that logs a warning -- "Skipping currently unsupported resource: /path/to/foo"?